src/libraries/System.Collections/src/System/Collections/BitArray.cs

```diff
@@ -144,7 +138,7 @@ public unsafe BitArray(bool[] values)
 
             if (Vector256.IsHardwareAccelerated)
             {
-                for (; (i + Vector256ByteCount) <= (uint)values.Length; i += Vector256ByteCount)
+                for (; (i + Vector256<byte>.Count) <= (uint)values.Length; i += (uint)Vector256<byte>.Count)
```

Comment:
These should be swapped so that only `i` is incremented in the loop.
```suggestion
                for (; i <= (uint)values.Length - Vector256<byte>.Count; i += (uint)Vector256<byte>.Count)
```
below the same.